docs/web3.contract.rst

```diff
@@ -809,6 +806,69 @@ take any arguments. For example:
         >>> myContract.functions.return13().call()
         13
 
+In cases where functions are overloaded and use arguments of similar types, a function
+may resolve to an undesired function when using the method syntax. For example, given
+two functions with the same name that take a single argument of differing types of
+``bytes`` and ``bytes32``. When a reference to :meth:`contract.functions.setBytes(b'1')`
+is used, the function will resolve as the one that takes ``bytes``. If a value is passed
+that was meant to be 32 bytes but the given argument was off by one, the function
+reference will still use the one that takes ``bytes``.
+
+When in doubt, use explicit function references to the signature. Use bracket notation
+(:meth:`contract.functions["setBytes(bytes32)"](b'')`) or the contract API
+`contract.get_function_by_signature("setBytes(bytes32)")` to retrieve the desired
+function. This will ensure an exception is raised if the argument is not strictly
+32 bytes in length.
+
+.. py:class:: ContractFunction
+
+Attributes
+~~~~~~~~~~
+
+The :py:class:`ContractFunction` class provides attributes for each function. Access the function attributes through `Contract.functions.myMethod`.
+
+.. py:attribute:: ContractFunction.myMethod(*args, **kwargs).abi_element_identifier
+
+    The signature of the function assigned to the class `__name__` during initialization.
```

Comment:
nit - seems like these should be double-tick (code-like) instead of single-tick (italics)?